packages/api-electron/index.js

```diff
@@ -127,47 +159,220 @@ const ready = (cb) => {
+  const [bounds] = args;
+  callWindowFunction(id, messageInfo, win => win.setBounds(bounds));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_ICON, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_BOUNDS, nonce];
+  const [icon] = args;
+  callWindowFunction(id, messageInfo, win => win.setIcon(icon));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_MAXIMIZABLE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_MAXIMIZABLE, nonce];
+  const [maximizable] = args;
+  callWindowFunction(id, messageInfo, win => win.setMaximizable(maximizable));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_MAXIMUM_SIZE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_MAXIMUM_SIZE, nonce];
+  const [width, height] = args;
+  callWindowFunction(id, messageInfo, win => win.setMaximumSize(width, height));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_MINIMIZABLE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_MINIMIZABLE, nonce];
+  const [minimizable] = args;
+  callWindowFunction(id, messageInfo, win => win.setMinimizable(minimizable));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_MINIMUM_SIZE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_MINIMUM_SIZE, nonce];
+  const [width, height] = args;
+  callWindowFunction(id, messageInfo, win => win.setMinimumSize(width, height));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_POSITION, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_POSITION, nonce];
+  const [x, y] = args;
+  callWindowFunction(id, messageInfo, win => win.setPosition(x, y));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_RESIZABLE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_RESIZABLE, nonce];
+  const [resizable] = args;
+  callWindowFunction(id, messageInfo, win => win.setResizable(resizable));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_SIZE, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_SIZE, nonce];
+  const [width, height] = args;
+  callWindowFunction(id, messageInfo, win => win.setSize(width, height));
+});
+
+ipc.on(IPC_SSF_SET_WINDOW_SKIP_TASKBAR, (e, id, nonce, args) => {
+  const messageInfo = [e.sender, IPC_SSF_SET_WINDOW_SKIP_TASKBAR, nonce];
+  const [skipTaskbar] = args;
+  callWindowFunction(id, messageInfo, win => win.setSkipTaskbar(skipTaskbar));
+});
+
+ipc.on(IPC_SSF_SHOW_WINDOW, (e, id, nonce) => {
+  const messageInfo = [e.sender, IPC_SSF_SHOW_WINDOW, nonce];
+  callWindowFunction(id, messageInfo, win => win.show());
+});
+
+ipc.on(IPC_SSF_MAXIMIZE_WINDOW, (e, id, nonce) => {
+  const messageInfo = [e.sender, IPC_SSF_MAXIMIZE_WINDOW, nonce];
+  callWindowFunction(id, messageInfo, win => win.maximize());
+});
+
+ipc.on(IPC_SSF_MINIMIZE_WINDOW, (e, id, nonce) => {
+  const messageInfo = [e.sender, IPC_SSF_MINIMIZE_WINDOW, nonce];
+  callWindowFunction(id, messageInfo, win => win.minimize());
+});
+
+ipc.on(IPC_SSF_UNMAXIMIZE_WINDOW, (e, id, nonce) => {
```

Comment:
We should probably split up these ipc events into another file and require them in here. Probably for another PR though